spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SparkZOrderStrategy.java

```diff
@@ -202,17 +202,14 @@ public Set<DataFile> rewriteFiles(List<FileScanTask> filesToRewrite) {
       tableCache().add(groupID, table());
       manager().stageTasks(table(), groupID, filesToRewrite);
 
-      // Disable Adaptive Query Execution as this may change the output partitioning of our write
-      SparkSession cloneSession = spark().cloneSession();
-      cloneSession.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), false);
-
+      SparkSession spark = spark();
```

Comment:
instead of cloning again, Aren't we suppose to store `spark`  coming in the constructor into a variable and use it here?

Reply:
this is not cloning but `spark()` from `SparkSortStrategy`

Reply:
Got it. This similar method name is really confusing. Just by reading this block of the code couldn't figure out spark() was also there in `SparkSortStrategy` and it is not from `RewriteDataFilesSparkAction`

Reply:
Maybe we can add a comment that it is from the parent or rename the method name in the parent.

Reply:
Sure, I'm wondering whether `spark` can be a protected field. The current way is a bit overuse of design patterns, IMO.

Reply:
Our current style rules don't allow for protected fields, but we could make an exception here I guess. I think long term we imagined that ZOrderStrategy will just be a special case of SortStrategy once we have the abilty to expression a multi arg transform as a Sort Order.